spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/outbound/HazelcastCacheWritingMessageHandler.java

```diff
@@ -16,64 +16,147 @@
 
 package org.springframework.integration.hazelcast.outbound;
 
-import java.util.List;
-import java.util.Map;
-import java.util.Queue;
-import java.util.Set;
-
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
 import org.springframework.integration.handler.AbstractMessageHandler;
-import org.springframework.integration.hazelcast.HazelcastIntegrationDefinitionValidator;
+import org.springframework.integration.hazelcast.HazelcastHeaders;
 import org.springframework.messaging.Message;
 import org.springframework.util.Assert;
 
+import reactor.util.StringUtils;
+
 import com.hazelcast.core.DistributedObject;
 import com.hazelcast.core.IList;
 import com.hazelcast.core.IMap;
 import com.hazelcast.core.IQueue;
 import com.hazelcast.core.ISet;
+import com.hazelcast.core.ITopic;
+import com.hazelcast.core.MultiMap;
+import com.hazelcast.core.ReplicatedMap;
 
 /**
- * MessageHandler implementation that writes {@link Message} payload to defined Hazelcast
- * distributed cache object. Currently, it supports {@link java.util.Map},
- * {@link java.util.List}, {@link java.util.Set} and {@link java.util.Queue} data
- * structures.
+ * MessageHandler implementation that writes {@link Message} or payload to defined
+ * Hazelcast distributed cache object.
  *
  * @author Eren Avsarogullari
  * @since 1.0.0
  */
 public class HazelcastCacheWritingMessageHandler extends AbstractMessageHandler {
 
-	private final DistributedObject distributedObject;
+	private DistributedObject distributedObject;
+
+	private String cacheExpression;
+
+	private String keyExpression;
 
-	public HazelcastCacheWritingMessageHandler(DistributedObject distributedObject) {
+	private boolean extractPayload = true;
+
+	public void setDistributedObject(DistributedObject distributedObject) {
 		Assert.notNull(distributedObject, "'distributedObject' must not be null");
 		this.distributedObject = distributedObject;
 	}
 
+	public void setCacheExpression(String cacheExpression) {
```

Comment:
No, according to the change to the `HazelcastOutboundChannelAdapterParser` this option must be of `Expression` type. As well as the next one.
It's really enough hard operation to parse expression at runtime.